feature: Adds columns to KeySelector

batture-rat opened the pull request:
resolves #10

This adds each subkey's last write time, as well as its subkey counts, to new cells in 
the `KeySelector` table widget. This is some bare-minimum useful information to 
have available when navigating the list. 

Also adds sorting on Key name, last write time, and subkey count in the
`KeySelector` widget. Decided to do this at the same time as https://github.com/orogenicbelt/hiview/issues/10 since
it also required changing the column highlight, and it became clear that
things weren't structured well as I continued to write this feature.
Keys/Subkeys are now wrapped in structs that provide a `From` trait
implementation for converting to their respective `Table`/`Row` ratatui
UI components.

@orogenicbelt I ended up knocking out two issues during development, but I'm
happy to split this into two branches and PRs if you prefer.

batture-rat committed:
feature: Adds columns to KeySelector

refs #10

This adds each subkey's last write time, as well as its subkey counts,
to new cells in the `KeySelector` table widget. This is some
bare-minimum useful information to have available when navigating the
list. All three of the cells in the `KeySelector` table should be
sortable when we implement sorting as a part of #5.

This also adds some refactoring to the `key_selector` module. Lots of
logic that was aggregating inside of the `render` method has been moved
out into structs wrappers and implementations that implement the `From`
trait between the wrappers and the UI components.

Keys/Subkeys are now wrapped in structs that provide a `From` trait
implementation for converting to their respective `Table`/`Row` ratatui
UI components.

orogenicbelt commented:
> @orogenicbelt I ended up knocking out two issues during development, but I'm
> happy to split this into two branches and PRs if you prefer.

The changes sound great and I like it all under one PR!

batture-rat committed:
feature: implements sorting on KeySelector cols

refs #5

Adds sorting on Key name, last write time, and subkey count in the
`KeySelector` widget. Decided to do this at the same time as #10 since
it also required changing the column highlight, and it became clear that
things weren't structured well as I continued to write this feature.
